### ISSUE TITLE
Distributions: impossible to hide bars and get just smooth kernel density lines

### ISSUE THREAD

**Issue by prdubois** [biolab/orange3#4119]
With the updated distributions widget, it seems impossible to get the view that is shown in the documentation:
![image](https://user-images.githubusercontent.com/8712266/67105469-d4daac80-f196-11e9-908a-b1988b879d77.png)

The closest I can get is this but I can't get rid of the bars:
![image](https://user-images.githubusercontent.com/8712266/67105600-0e131c80-f197-11e9-8133-e68b7437cb51.png)

Any suggestion?

I'm using v3.23.1

**Comment by janezd** [comment]
Distributions widget has been redesigned from scratch. It should probably also be renamed to Histogram. Documentation often lags behind, like here.

You can no longer remove the bars, but this is by design. The widget is a histogram, which can *also* fit some curves to the data. The curves themselves, especially non-parametric curves, can be misleading, so it is better, imho, to keep the histogram visible.

**Comment by prdubois** [comment]
I disagree. Yes, sometimes hiding the bars could be misleading but keeping them makes for an extremely busy and unreadable graph with more than a few variables. The original distribution widget was one of the most useful widget for data exploration in Orange. I used it many times to wow colleagues with the nice graphs it made. Now that I can't remove the bars, I can't really use it anymore for "presentation quality" graphs.

Let me show you a concrete example:

Before (legend edited for confidential reasons):
![image](https://user-images.githubusercontent.com/8712266/67287946-0f4f8c80-f4aa-11e9-8c39-e736e7da43c8.png)

After (the input data is slightly different)
![image](https://user-images.githubusercontent.com/8712266/67288159-710ff680-f4aa-11e9-8c64-9412dc22b7af.png)

The bars here add nothing useful in this case, it's just pollution. 

Also the filled area under the curves added a nice professional touch, I'm sad that it's gone.

One last thing which should probably have its own separate issue is that the legend doesn't work as well anymore. I have an upstream Select Row widget to only keep 7 categories (the 7 lines plotted) but the legend uses all the unfiltered categories.

**Comment by ajdapretnar** [comment]
Your data seems to be time series, so it might be best to use Timeseries add-on for plotting such data. Moreover, I would say histograms look much more informative than distributions.
![plot](https://user-images.githubusercontent.com/12524972/67289452-eb5a6e00-f4de-11e9-8e04-c304d59945a2.png)
Please correct me if I am wrong.

As for the second part - this is just a change in setting. Tick both boxes in Purging and you will see the same results as before.

**Comment by prdubois** [comment]
Hello,
My data is indeed time based but it's not a time series. The X axis is "Months to failure". It's from analysis of instruments failures in the field. So the time series add-on is not really appropriate here.

I disagree with the statement that histograms are more informative (at least not always). My goal here is to show a general trend, not get exact numbers. It's much easier for the eye to follow a curve than to try to make sense of several vertical bars. Again with my example, which one is easier to understand?

![image](https://user-images.githubusercontent.com/8712266/67317508-2d33e600-f4d8-11e9-8bcd-471fc5768945.png)

or (minus the bars)
![image](https://user-images.githubusercontent.com/8712266/67317467-1b524300-f4d8-11e9-8c13-c2521b2a7d29.png)


Both types have value. Declaring that "bars are better in every single cases therefore they should always be ON" is shortsighted in my opinion. Give the user the flexibility please.

Thanks for the purging pointer. It indeed works.

**Comment by janezd** [comment]
Filled areas under curves indeed looked nice. I guess we should resuscitate this.

> Again with my example, which one is easier to understand?

None, in my opinion. :)

The problem with bars is that you plotted too many. When I teach data mining, I use this as an example of overfitting. There is not enough data to support this many bars. Decrease the number of bars and the image will become nice and understandable. Your example is not fair.

The problem with curves is that they suggest that, say, the failure times for the blue points are three-modal, which is probably not the case. The problem is the same as with the bars -- the kernel is too narrow for the sample size. (I also use this, just with even narrower kernels, as an example of overfitting -- I later refer to this in lessons on predictive modelling.) You should widen the kernel or, perhaps better, use some parametric curve.

The renewed widget initially didn't even have non-parametric curves because their shapes depended on the kernel width and the user has no principled way to set it. (In classification we have a similar problem, but there we can see how different regularization strengths affect the performance, while here we can't.)

Whatever the curve's shape and hyperparameters, it masks the underlying data and can present a wrong picture. Even with parametric curves, you could choose, say, Gaussian curves when your data is clearly bimodal, and without seeing the bars you would never know it.

I can however agree with your argument about aesthetics. If the curves are done right, they can convey the correct information without the bars. You have a point there.

I'm reopening the issue and adding a "Needs Discussion" tag. This marks it for discussion on next Friday's meeting of the core team.

**Comment by prdubois** [comment]
Good point. You're right, my example wasn't fair, I used the most bars on purpose to convey my point.

Here's the same data with a bin width that makes more sense (v3.23.1)
![opt3](https://user-images.githubusercontent.com/8712266/67348614-7190a780-f513-11e9-9907-75a1b1953b08.png)

And here's how the same dataset looks like with v3.18 (not sure when the change was introduced), which in my opinion conveys the same information but is more aesthetically pleasing and easier to understand:
![opt4](https://user-images.githubusercontent.com/8712266/67348626-79504c00-f513-11e9-8241-d112d19203b2.png)

In any case, thanks for reconsidering my request.

**Comment by janezd** [comment]
@prdubois, I have good news for you. :)

We'll (probably, almost certainly) add a checkbox for hiding the bars, though we're not yet sure where. UX will be a bit tricky.

**Comment by ramadap** [comment]
What about relative frequency? That's gone as well :(

**Comment by ajdapretnar** [comment]
It's still there and it's called show probabilities. Someone didn't update the documentation yet! 😒

**Comment by ramadap** [comment]
Thanks Adja, but it's not the same... previously the adjustment was to the size of each group (group by option in the previous version, now, split by), now it just normalizes the whole graph to 1. Or am I missing something?

**Comment by ajdapretnar** [comment]
You can use Box Plot for that.

Also, not Adja, but Ajda.

**Comment by ramadap** [comment]
@ajdapretnar so sorry about the misspelling, Ajda.

Yep, I know about boxplot, but I need binning, which boxplot doesn't do. 

Is there anyway we can reinstall the older distribution widget on the newest version?

**Comment by ajdapretnar** [comment]
I am almost absolutely sure relative frequency was from 0 to 1 in the previous version. Because that's what relative means. Do you have any example of what you wish to achieve?

No, you cannot use the old version of the widget in the newer Orange.

**Comment by ramadap** [comment]
Two groups, absolute vs relative frequency. This was useful but apparently lost in the latest version update.


<img width="1600" alt="absolute" src="https://user-images.githubusercontent.com/3635823/71138708-3d302380-2204-11ea-9b35-7656e360dd33.png">
<img width="1600" alt="relative" src="https://user-images.githubusercontent.com/3635823/71138716-43260480-2204-11ea-9303-d3e7dc375093.png">

**Comment by janezd** [comment]
#4301 adds hiding bars.

The weird normalization from the old incarnation won't be reimplemented. The widget was reimplemented from scratch precisely because it tried to do too many things at once, which made it a mess from both user's and programmer's perspective. The normalization it performed was difficult to explain; anytime I tried doing it while teaching I got lost myself together with students. More importantly, it doesn't really belong to this widget. It was replaced by "Show probabilities", which makes sense in this context. Adding the previous normalization (although this would be trivial), would clash with Show probabilities and put us back on the previous track.

**Comment by ramadap** [comment]
Paint me confused... what's so difficult about explaining relative frequencies? It's your system, but you just removed a useful feature.

**Comment by janezd** [comment]
But the current widget gives you exactly that -- it shows you relative frequencies. For each bin, you see proportions of data instances belonging to each group -- these are proper relative frequencies.

The previous incarnation scaled bar heights according to the group size, so that you could compare number of data instances in each bin if the total number of instances in each group were equal. These are indeed relative frequencies in some sense, but I'd rather call it "scaling by group size".
